code/src/App.js

```diff
@@ -1,9 +1,116 @@
-import React from 'react'
+import React,{ useState, useEffect } from 'react';
+
+import Header from './Component/Header';
+import ThoughtSubmitForm from 'Component/ThoughtSubmitForm';
+import MessageList from 'Component/MessageList';
+import { API_URL } from './constants';
+import './index.css';
+
+//there are 3 erros found in the response relating to empty, too short or too long messages.
+const THOUGHT_ERRORS = [
+  { kind: 'required', message: 'Please type in your happy thought' },
+  { kind: 'minlength', message: 'Message should be longer than 5 characters' },
+  { kind: 'maxlength', message: 'Oops! Can you keep the message shorter than 140 characters, please.' }
+];
 
 export const App = () => {
+  const [thought, setThought] = useState(""); //the thought state needs to be here in the App.js instead in the child component to handle the erro update while hitting submit button.
+  const [thoughts, setThoughts] = useState([]);
+  const [errorMessage, setErrorMessage] = useState(null);
+
+
+  useEffect(() => {
+    fetchMessages();
+  }, []);
+
+  const fetchMessages = () => {
+    fetch(API_URL)
+      .then((res) => {
+        return res.json();
+      })
+      .then((data) => {
+        setThoughts(data)
+      });
+  };
+
+  const handleThoughtChange = (newThought) => {
+    setThought(newThought);
+  };
+
+  // errorMassage will be cleared and Textarea for thought input should be cleared too.
+  const clearForm = () => {
+    setErrorMessage(null);
+    setThought("");
+  };
+
+  const handleThoughtSubmit = (thought) => {
+
```

Comment:
why the empty line here?

Reply:
Hehe, i dont know why. Removed space! Thanks!